### PR TITLE
[WIP] Introducing a Grid system (allow dynamic query/view on listing pages)

### PR DESCRIPTION
**Pull request by mickaelandrieu**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Introducing a Grid system (allow dynamic query/view on listing pages)
| Type?         | new feature
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Make listing pages dynamic and highly customizable, avoid useless duplications.
| How to test?  | well... it's a WIP or a POC

# This PR won't be merged, check #8986 instead

#### Important guidelines

* Make sure [your local branch is up to date](https://help.github.com/articles/syncing-a-fork/) before commiting your changes!
* Your code MUST respect [our Coding Standards](http://doc.prestashop.com/display/PS16/Coding+Standards) (for code written in PHP, JavaScript, HTML/CSS/Smarty/Twig, SQL)!
* Your commit name MUST respect our [naming convention](http://doc.prestashop.com/display/PS16/How+to+write+a+commit+message)!

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8987)

### CODE REVIEW

File: src/Core/Grid/Configuration/LogsConfiguration.php [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            ['name' => 'ID'],
+            ['name' => 'Employee'],
+            ['name' => 'Severity (1-4)'],
+            ['name' => 'Message'],
+            ['name' => 'Object type'],
+            ['name' => 'Error code'],
+            ['name' => 'Date', 'type' => 'date_interval']
+        ]);
+
+        $this->hookDispatcher->dispatchForParameters("change{$this->name}Columns", [
```
Comment (kpodemski) [inline src/Core/Grid/Configuration/LogsConfiguration.php:74]: 👍

File: src/PrestaShopBundle/Resources/config/services/grids.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+services:
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Resources/config/services/grids.yml:1]: @sarjon wdyt about having configurable tables using yml?

**Comment by mickaelandrieu** [comment]
To be honest, I prefer the following implementation => https://github.com/PrestaShop/PrestaShop/pull/8986

Some things are missing, but this can be improved!

I keep the pull request opened to allow the comparaison, but imo we should invest time on @sarjon's work.

**Comment by Quetzacoalt91** [comment]
Did you mean #8986?

clippy

File: src/Core/Grid/Configuration/ConfigurationInterface.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+     * @return array
+     */
+    public function getActions();
+
+    /**
+     * The list of available bulk actions in Grid.
+     *
+     * @return array
+     */
+    public function getBulkActions();
```
Comment (mickaelandrieu) [inline src/Core/Grid/Configuration/ConfigurationInterface.php:55]: @sarjon should the Table/Grid also manage bulk action and how?

File: src/Core/Grid/Configuration/LogsConfiguration.php [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        return $this->name;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getColumns()
+    {
+        $columns = ColumnCollection::createFromArray([
+            ['name' => 'ID'],
```
Comment (mickaelandrieu) [inline src/Core/Grid/Configuration/LogsConfiguration.php:65]: another question: how can we manage labels with different translation domain names?

File: src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/LogsController.php [hunk 5]
```diff
@@ -33,6 +33,8 @@
 use Symfony\Component\HttpFoundation\RedirectResponse;
 use PrestaShopBundle\Security\Voter\PageVoter;
 use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Cookie;
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/LogsController.php:37]: forget about it, Search won't use cookies but db

File: src/PrestaShopBundle/Entity/Repository/LogRepository.php [hunk 6]
```diff
@@ -55,6 +56,14 @@ public function findAll()
         return $statement->fetchAll();
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function getItems()
+    {
+        return $this->findAll();
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Entity/Repository/LogRepository.php:64]: @sarjon we may make consistent the way we retrieve data, some thing like itemsCollection to ensure the table will always be rendered, wdyt?

**Pull request closed without merging**